### ISSUE TITLE
Why need to set retain_graph=True when calling .backward()?

### ISSUE THREAD

**Issue by xiami2019** [andy194673/joust#4]
Hi, nice work! 
I wonder that why using "update_loss.backward(retain_graph=True)" in the update function. However, it seems that there is no need to retain computation graph since every loss just need calling backward() once.  Looking for your reply.
